herddb-core/src/main/java/herddb/sql/CalcitePlanner.java

```diff
@@ -505,6 +506,7 @@ private PlannerResult runPlanner(String defaultTableSpace, String query) throws
         RelDataType originalRowType = logicalPlan.getRowType();
         RelOptCluster cluster = logicalPlan.getCluster();
         final RelOptPlanner optPlanner = cluster.getPlanner();
+        optPlanner.addRule(ReduceExpressionsRule.FILTER_INSTANCE);
```

Comment:
@eolivelli what is the purpose of adding the Filter_Instance here?

Reply:
This is the core of this change.
That Rule simplifies every expression, removing redundant filterd

Reply:
I see. I think this looks good to me.